### PR TITLE
UCS/TEST: Implement a page table data structure for address intervals.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1225/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/640/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1226/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/641/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1227/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1230/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/643/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@miked-mellanox @shamisp @alex-mikheev  pls review

File: src/ucs/datastruct/pgtable.c [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+#define ucs_pgt_entry_get_region(_pte) \
+    ({ \
+        ucs_assert(ucs_pgt_entry_test(_pte, UCS_PGT_ENTRY_FLAG_REGION)); \
+        (ucs_pgt_region_t*)ucs_pgt_entry_value(_pte); \
+    })
+
+#define ucs_pgt_entry_get_dir(_pte) \
+    ({ \
+        ucs_assert(ucs_pgt_entry_test(_pte, UCS_PGT_ENTRY_FLAG_DIR)); \
+        (ucs_pgt_dir_t*)ucs_pgt_entry_value(_pte); \
```
Comment (miked-mellanox) [inline src/ucs/datastruct/pgtable.c]: should be FLAG_DIR mask applied here?

Comment (yosefe) [inline src/ucs/datastruct/pgtable.c]: ucs_pgt_entry_value() macro applies this flag

Comment (miked-mellanox) [inline src/ucs/datastruct/pgtable.c]: what is a difference (beside assert) with ucs_pgt_entry_get_region(_pte) ?

Comment (yosefe) [inline src/ucs/datastruct/pgtable.c]: casting the unsigned long to ucs_pgt_dir_t*

**Comment by shamisp** [comment]
autobot build

File: src/ucs/datastruct/pgtable.c (hunk 1 above)
Comment (miked-mellanox) [inline src/ucs/datastruct/pgtable.c]: ohh. right, thx

**Comment by shamisp** [comment]
@sameerkm please take a look

**Comment by shamisp** [comment]
:+1:

**Comment by miked-mellanox** [comment]
:+1:

**Pull request merged**